lib/vm/src/trap/traphandlers.rs

```diff
@@ -894,14 +895,12 @@ fn on_wasm_stack<F: FnOnce() -> T, T>(
     // allows them to be reused multiple times.
     // FIXME(Amanieu): We should refactor this to avoid the lock.
     lazy_static::lazy_static! {
-        static ref STACK_POOL: Mutex<Vec<DefaultStack>> = Mutex::new(vec![]);
+        static ref STACK_POOL: crossbeam_queue::SegQueue<DefaultStack> = crossbeam_queue::SegQueue::new();
     }
     let stack = STACK_POOL
-        .lock()
-        .unwrap()
         .pop()
         .unwrap_or_else(|| DefaultStack::new(stack_size).unwrap());
-    let mut stack = scopeguard::guard(stack, |stack| STACK_POOL.lock().unwrap().push(stack));
+    let mut stack = scopeguard::guard(stack, |stack| STACK_POOL.push(stack));
```

Comment:
This is not directly related, but should we use an upper bound for the number of stored stack here?
For example, if you had very bursty workloads and only ran a lot of instances for a short time, while mostly remaining idle, we'd have lots of redundant memory usage.

Happy to also just create a followup issue to discuss / track this.

Reply:
The `ArrayQueue` actually makes this possible, so it'd be simple to implement here. However, we'd want to make something like that configurable, so it'll be bigger in scope and we can do that separately.